Adding docker_service

chouseknecht opened the pull request:
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

docker_service.py
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel 30decd5e0a) last updated 2016/05/10 09:39:13 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 1f5cf669dd) last updated 2016/05/06 13:36:42 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 431591c2b4) last updated 2016/05/06 13:34:47 (GMT -400)
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

New module based on docker/compose python module. Consumes docker compose from a project directory or inline, allowing for the management of one or more services and containers in a stand-alone or swarm environment. Supports check mode and differences and attempts to idempotent.

Commit:
Adding docker_service

chouseknecht commented:
Did not realize the original `docker_service` branch was created in upstream. Moved back to my fork. 

Set version_added to 2.2 so that test run; however, the intention is to get this into 2.1.

gregdek commented:
Thanks @chouseknecht for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

Commit:
Removed fail method. Combined recreate and force_recreate params.

Commit:
Removed use of --diff. Replaced 'diff' in output with 'actions'. Only show 'actions' in output if debug true.

chouseknecht commented:
- Removed the fail() method
- Collapsed 'recreate' and 'force_recreate' options
- Removed use of --diff and 'diff' in output. Renamed 'diff' output to 'actions', and only including 'actions' when debug on.

Commit:
Added actions to RETURN docs. Show actions when debug or check mode.

Commit:
Moving version added to 2.1
